sound/soc/sof/ipc4-topology.h

```diff
@@ -148,13 +148,17 @@ struct ipc4_pipeline_set_state_data {
  * struct sof_ipc4_available_audio_format - Available audio formats
  * @base_config: Available base config
  * @out_audio_fmt: Available output audio format
+ * @input_audio_fmts: Available input audio formats
  * @ref_audio_fmt: Reference audio format to match runtime audio format
  * @dma_buffer_size: Available Gateway DMA buffer size (in bytes)
- * @audio_fmt_num: Number of available audio formats
+ * @audio_fmt_num: Number of available audio formats. Audio formats defined in topology always
+ * contain both the input and output formats for a particular pin. So audio_fmt_num is the same
+ * for both available input and output formats.
```

Comment:
this makes no sense to me since a pin is source or sink only, not both.  A 'particular pin' cannot have both 'input and output formats'.

Reply:
I agree but this is the path or least resistance to resuse the audio format definitions in the current tplg. We've defined them together in the same set tokens for pin 0. CHanging this would require a huge amount of changes in the tplg. But if we should fix it, now would be the time even if it is painful. What do you think?

Reply:
I don't understand what the topology definitions might be, we haven't introduced pins just yet?

Or maybe we are confused in that there can be a sink pin0 and a source pin0, they have the same index but are NOT the same pin.

Reply:
There are no pins in topology yet but this is how an audio format object is defined. It includes the audio formats for both input and output pins (pin 0 implicitly)

Reply:
oh well. I think we need to fix this, possibly with a carefully planned step to avoid hitting milestones, but conceptually something like a SRC or a downmix cannot have the same formats on input and output by construction. 
Better fix this now than build our own legacy that we have to work-around later.

Reply:
but @plbossart this definition doesnt assume same formats for input and output. It just assumes that they have to come in pairs.

Reply:
I do have a simpler fix in tplg. Let me post it

Reply:
it's still problematic, take SRC where the input pin could support all sampling rates but the output only 48kHz for example.
Same for the downmix, we could have different channel maps on the input and plain old stereo on the output.

And we know there will be modules without output (acoustic event detection) or without input (tone generator)

Reply:
I just created this PR to fix this problem: https://github.com/thesofproject/sof/pull/7014

Reply:
The the second from last patch in this series handles the 2 new tokens from the SOF PR. @plbossart let me know how it looks now